ibis-server/app/model/metadata/athena.py

```diff
@@ -0,0 +1,139 @@
+        # which does not work with information_schema queries.
+        cursor = self.connection.raw_sql(sql)
+        response = pd.DataFrame(
+            cursor.fetchall(), columns=[col[0] for col in cursor.description]
+        ).to_dict(orient="records")
+
+        def get_column(row) -> Column:
+            return Column(
+                name=row["column_name"],
+                type=self._transform_column_type(row["data_type"]),
+                notNull=row["is_nullable"].lower() == "no",
+                description="",  # Athena doesn't provide column descriptions in information_schema
+                properties=None,
+            )
+
+        def get_table(row) -> Table:
+            return Table(
+                name=self._format_athena_compact_table_name(
+                    row["table_schema"], row["table_name"]
+                ),
+                description="",  # Athena doesn't provide table descriptions in information_schema
+                columns=[],
+                properties=TableProperties(
+                    schema=row["table_schema"],
+                    catalog=row["table_catalog"],
+                    table=row["table_name"],
+                ),
+                primaryKey="",
+            )
+
+        unique_tables = {}
+
+        for column_metadata in response:
+            # generate unique table name
+            table_name = self._format_athena_compact_table_name(
+                column_metadata["table_schema"], column_metadata["table_name"]
+            )
+            # init table if not exists
+            if table_name not in unique_tables:
+                unique_tables[table_name] = get_table(column_metadata)
+
+            current_table = unique_tables[table_name]
+            # add column to table
+            current_table.columns.append(get_column(column_metadata))
+
+        return list(unique_tables.values())
+
+    def get_constraints(self) -> list[Constraint]:
+        # Athena doesn't support foreign key constraints
+        return []
+
+    def get_version(self) -> str:
+        return "AWS Athena - Follow AWS service versioning"
+
+    def _format_athena_compact_table_name(self, schema: str, table: str) -> str:
+        return f"{schema}.{table}"
+
+    def _transform_column_type(self, data_type):
+        data_type = re.sub(r"\(.*\)", "", data_type).strip()
+        switcher = {
+            # String Types (ignore Binary and Spatial Types for now)
+            "char": RustWrenEngineColumnType.CHAR,
+            "varchar": RustWrenEngineColumnType.VARCHAR,
+            "tinytext": RustWrenEngineColumnType.TEXT,
+            "text": RustWrenEngineColumnType.TEXT,
+            "mediumtext": RustWrenEngineColumnType.TEXT,
+            "longtext": RustWrenEngineColumnType.TEXT,
+            "enum": RustWrenEngineColumnType.VARCHAR,
+            "set": RustWrenEngineColumnType.VARCHAR,
+            # Numeric Types(https://dev.mysql.com/doc/refman/8.4/en/numeric-types.html)
```

Comment:
Is the comment outdated? It looks like a MySQL link 🤔

Reply:
Thanks for reviewing. I’ve removed the unrelated link 😅 and added the Glue Catalog test case.